src/dev/flang/be/interpreter/Executor.java

```diff
@@ -533,11 +533,11 @@ public Pair<Value, Object> match(int s, AbstractInterpreter<Value, Object> ai, V
     var cix = _fuir.matchCaseIndex(s, tagAndChoiceElement.v0());
 
     var field = _fuir.matchCaseField(s, cix);
-    if (field != -1)
+    if (field != -1 && !_fuir.clazzIsUnitType(_cur.clazz()))
```

Comment:
I think this should work as well and it would work even if `_cur.clazz()` is not a unit type due to some other fields unrelated to the `match`:
```suggestion
    if (field != -1 && !_fuir.clazzIsUnitType(_fuir.clazzResultClazz(field)))
```